workspace-service/src/graphql/folders.rs

```diff
@@ -86,12 +88,14 @@ impl FoldersMutation {
         let pool = context.data()?;
         let workspace = Uuid::parse_str(&new_folder.workspace)?;
         let event_client: &EventClient = context.data()?;
+        let requesting_user = context.data::<super::RequestingUser>()?;
```

Comment:
Since you import `RequestingUser` above, you don't need `super::` here. You can also put the type behind the variable for consistency with the line above.

Thinking about it, I think the type annotation isn't needed at all. The type should be clear because we're passing `requesting_user` into `create_folder`. The same goes for the event client. I think you can remove the type annotation there, too.

```suggestion
        let requesting_user = context.data()?;
```